estate/models/estate_property_offer.py

```diff
@@ -15,3 +20,34 @@ class EstatePropertyOffer(models.Model):
     )
     partner_id = fields.Many2one("res.partner", required=True)
     property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline",
+        inverse="_inverse_date_deadline",
+        string="Deadline",
+    )
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            if not record.create_date:
+                record.date_deadline = datetime.today() + relativedelta(days=record.validity)
```

Comment:
To avoid import datetime and relativedelta you could use `fields.Date.add()` method you can read about it [here](https://github.com/odoo/odoo/blob/16.0/odoo/fields.py#L2021)

```suggestion
                record.date_deadline = fields.Date.add(fields.Date.today(), days=record.validity)
```

Apply it to the else sentence too and remove the imports please.